#!/usr/bin/env python
PACKAGE = "moving_object_tracker"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("covariance_trace_limit", double_t, 0, "Limit of covariance value to remove uncorrected tracker", 0.5, 0.0, 1.5)
gen.add("correction_count_limit", int_t, 0, "If correction count is under this value, object is not published", 3, 1, 15)
gen.add("object_radius", double_t, 0, "Object radius used to generate new trackers", 0.5, 0.0, 2.0)
#gen.add("gating_mahalanobis", double_t, 0, "Gating threshold by mahalanobis distance", 3.0, 0.0, 10.0)
#gen.add("gating_deviation", double_t, 0, "Gating threshold by deviation", 1.5, 0.0, 5.0)

exit(gen.generate(PACKAGE, "moving_object_tracker", "MovingObjectsTracker"))
